src/App.tsx

```diff
@@ -77,28 +96,43 @@ export const App: React.FC = () => {
     setDetails(data.marketDetails)
   }
 
+  const findSelectedMarket = (marketID: number) => {
+    setSelectedMarket(marketDetails.find(market => market.id === marketID))
+    history.push(`/markets/${marketID}`);
+  }
   return (
     <>
       <ScrollToTop />
 
       <header>
         <h1>Freshly Fetched</h1>
       </header>
-      <main>
-        <Switch>
-          <Route
-            exact
-            path='/'
-            render={() => <Search getMarkets={getMarkets} />}
-          />
-          <Route
-            exact
-            path='/markets'
-            render={() => (
-              <Results
-                allMarkets={allMarkets}
-                marketDetails={marketDetails}
-                zip={zip}
+      <Switch>
+        <Route
+          exact
+          path='/'
+          render={() => <Search getMarkets={getMarkets} />}
+        />
+        <Route
+          exact
+          path='/markets'
+          render={() => (
+            <Results
+              allMarkets={allMarkets}
+              zip={zip}
+              findSelectedMarket={findSelectedMarket} 
+            />
+          )}
+        />
+        <Route
+          exact
+          path='/markets/:id'
+          render={({ match }) => {
+            const { id } = match.params;
+            return (
+              <Details
+                id={id}
+                selectedMarket={selectedMarket}
```

Comment:
Nice! So this Route is in lieu of the one below? Do we still need both? If we do, can we refactor these two Routes together somehow?